### ISSUE TITLE
fairseq-train not utilizing GPU

### ISSUE THREAD

**Issue by babangain** [pytorch/fairseq#2626]
When I try to train a model using fairseq-train, it trains on CPU only. And uses up to 2000% on a server. But never uses GPU. Even though other applications are Using GPU successfully. 

    Environment: Conda 4.8.3
    Python: 3.8.3
    Torch: 1.4.0+cu100
    OS: Ubuntu 18.04.1 LTS

Full command:

``
CHECKPOINT_DIR=checkpoints_de_en_parallel
nohup fairseq-train \
    data-bin/wmt18_de_en \
    --source-lang de --target-lang en \
    --arch transformer_wmt_en_de_big --share-all-embeddings --log-interval  1 \
    --dropout 0.3 --weight-decay 0.0 --log-format simple \
    --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
    --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
    --lr 0.001 --lr-scheduler inverse_sqrt --warmup-updates 4000 --num-workers 1 \
    --max-tokens 500 --update-freq 2 \
    --max-update 30000 \
    --save-dir $CHECKPOINT_DIR &
``

**Comment by gvskalyan** [comment]
Please check if you have installed torch such that it supports gpu.
In your python environment, check
```python
import torch
torch.cuda.is_available()
torch.cuda.device_count()
```

**Comment by babangain** [comment]
torch.cuda.is_available()  >>> True
    torch.cuda.device_count() >>> 7

**Comment by gvskalyan** [comment]
Is it supposed to be 7 or 8 GPUs, Can you please try with one GPU using CUDA_VISIBLE_DEVICES=1 with fairseq-train

**Comment by babangain** [comment]
Yes, It was supposed to be 7 GPU.
and similar fate with  CUDA_VISIBLE_DEVICES=1
I resolved the problem now. 

I think It was a mismatch between CUDA versions. I installed PyTorch 1.2 using pip. And my CUDA version is 10.0.13. But earlier, I installed PyTorch 1.6 for CUDA 10.1 which was unsupported. After installing the correct version, even though torch.cuda.is_available() and other were showing properly, I don't know why fairseq-train was not using CPU. I tried uninstalling torch fairseq. ( from pip)
Then tried to install torch via conda. There was some resolve environment problem..

So, I created a new env. 
Installed pytorch 1.2 cudatoolkit 10.0 
then installed fairseq using pip. 
And it's using GPU now.

Thank you @gvskalyan anyway.
